src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php

```diff
@@ -507,6 +508,30 @@ public function getWriteInfo(string $class, string $property, array $context = [
         return $noneProperty;
     }
 
+    public function getAttributes(string $class, string $property, array $context = []): ?array
+    {
+        try {
+            $reflClass = new \ReflectionClass($class);
+        } catch (\ReflectionException) {
+            return null;
+        }
+
+        if (!$reflClass->hasProperty($property)) {
+            return null;
+        }
+
+        $attributes = [];
+        $reflProperty = $reflClass->getProperty($property);
+        foreach ($reflProperty->getAttributes() as $attribute) {
+            $attributes[] = [
+                'name' => $attribute->getName(),
+                'arguments' => $attribute->getArguments(),
+            ];
+        }
+
+        return $attributes;
```

Comment:
```suggestion

        return $reflProperty->getAttributes();
```

Why not returning attributes themselves? So that we can have all the attribute data (repeated, etc) and use the `newInstance` method.

If it was because you want to decouple `getAttributes` from `\ReflectionXXX`, I think that we shouldn't try to decouple as it's part of the PHP language itself.